sympy/core/function.py

```diff
@@ -1249,11 +1249,15 @@ def __new__(cls, expr, *variables, **assumptions):
                     old_v = v
                     v = new_v
                 obj = expr
-                for i in range(count):
-                    obj2 = deriv_fun(obj, v)
-                    if obj == obj2:
-                        break
-                    obj = obj2
+                try:
```

Comment:
Is it possible to handle the problem without `try`? It usually makes the code complicated to read.

Reply:
Should I use `if-else` statement ?

Reply:
I guess the exception is raised if `count` is a symbol or non-number.